### PR TITLE
[BEAM-4102] use plugins DSL instead of manual classpath

### PR DESCRIPTION
**Pull request by adude3141**
Change beams way of applying plugins

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by adude3141** [comment]
@kennknowles PTAL if it meets your intentions

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -16,23 +16,41 @@
...
+  maven { url "http://repo.spring.io/plugins-release" }
+}
 
+// Dependencies on other plugins used when this plugin is invoked
 dependencies {
   compile gradleApi()
   compile localGroovy()
   compile 'com.github.jengelman.gradle.plugins:shadow:4.0.3'
+
+  runtime "net.ltgt.gradle:gradle-apt-plugin:0.20"                                                  // Enable a Java annotation processor
```
Comment (kennknowles) [inline buildSrc/build.gradle]: Wow, this is much better to have these dependencies adjacent to the code that needs them. Great!

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -54,3 +72,20 @@ spotless {
     greclipse().configFile('greclipse.properties')
   }
 }
+
+gradlePlugin {
+  plugins {
+    beamModule {
+      id = 'org.apache.beam-module'
```
Comment (kennknowles) [inline buildSrc/build.gradle]: Ah, I didn't know it was so easy! When I was reading about it, I thought you had to actually publish the plugin somewhere in order to use it by id.

Nit: maybe `org.apache.beam.module` and `org.apache.beam.vendor-java` and `org.apache.beam.jenkins`.

**Comment by adude3141** [comment]
Resolved merge conflicts.

**Comment by adude3141** [comment]
Run Java PreCommit

**Comment by adude3141** [comment]
Run Spotless PreCommit

**Comment by adude3141** [comment]
retest this please

**Comment by kennknowles** [comment]
Nice and green!

**Comment by adude3141** [comment]
Thx for your help merging, @kennknowles

**Pull request merged**